keras/layers/convolutional.py

```diff
@@ -5,16 +5,15 @@
 import theano.tensor as T
 from theano.tensor.signal import downsample
 
-from .. import activations, initializations
+from .. import activations, initializations, regularizers, constraints
 from ..utils.theano_utils import shared_zeros
 from ..layers.core import Layer
 
-
 class Convolution1D(Layer):
     def __init__(self, input_dim, nb_filter, filter_length,
         init='uniform', activation='linear', weights=None,
         border_mode='valid', subsample_length=1,
-        W_regularizer=None, b_regularizer=None, activity_regularizer=None, W_constraint=None, b_constraint=None):
+        W_regularizer='identity', b_regularizer='identity', activity_regularizer='identity', W_constraint='identity', b_constraint='identity'):
```

Comment:
For efficiency reasons I believe it is preferable to keep the default regularizers to `None`. This is also important because using `None` in the class constructor is part of the standard API and the class should be able to deal with it.

This is absolutely not an issue in `get_config`: you can simply use, e.g.:
```python
"W_constraint":self.W_constraint.get_config() if  self.W_constraint else None,
```
After setting `self.W_constraint` to the value given to the constructor in `__init__`.